src/tests/generatekey.cpp

```diff
@@ -650,11 +650,14 @@ TEST_F(rnp_tests, rnpkeys_generatekey_testExpertMode)
 {
     cli_rnp_t rnp;
     rnp_cfg_t ops = {0};
+    rnp_cfg_t tops = {0};
 
-    assert_true(rnp_cfg_setbool(&ops, CFG_EXPERT, true));
-    assert_true(rnp_cfg_setint(&ops, CFG_S2K_ITER, 1));
+    assert_true(rnp_cfg_setbool(&tops, CFG_EXPERT, true));
+    assert_true(rnp_cfg_setint(&tops, CFG_S2K_ITER, 1));
 
     /* ecdsa/ecdh p256 keypair */
+    rnp_cfg_free(&ops);
+    rnp_cfg_copy(&ops, &tops);
     assert_true(rnp_cfg_setstr(&ops, CFG_USERID, "expert_ecdsa_p256"));
```

Comment:
It would be better instead of modifying the whole test workflow just inset `rnp_cfg_unset()` calls. However, anyway, you should not modify inner code of `rnp_cfg_setstr()` and use `rnp_cfg_addstr()` or something similar.